src/anomalib/data/image/mvtec_loco.py

```diff
@@ -0,0 +1,466 @@
+
+        .. code-block:: python
+
+            dataset.task = "segmentation"
+            dataset.setup()
+            print(dataset[0].keys())
+            # Output: dict_keys(['image_path', 'label', 'image', 'mask_path', 'mask'])
+
+        The image is a torch tensor of shape (C, H, W) and the mask is a torch tensor of shape (H, W).
+
+        .. code-block:: python
+
+            print(dataset[0]["image"].shape, dataset[0]["mask"].shape)
+            # Output: (torch.Size([3, 256, 256]), torch.Size([256, 256]))
+    """
+
+    def __init__(
+        self,
+        task: TaskType,
+        root: Path | str = "./datasets/MVTec_LOCO",
+        category: str = "breakfast_box",
+        transform: Transform | None = None,
+        split: str | Split | None = None,
+    ) -> None:
+        super().__init__(task=task, transform=transform)
+
+        self.root_category = Path(root) / category
+        self.split = split
+        self.samples = make_mvtec_loco_dataset(
+            self.root_category,
+            split=self.split,
+            extensions=IMG_EXTENSIONS,
+        )
+
+    def __getitem__(self, index: int) -> dict[str, str | torch.Tensor]:
+        """Get dataset item for the index ``index``.
+
+        This method is mostly based on the super class implementation, with some different as follows:
+            - Using 'torch.where' to make sure the 'mask' in the return item is binarized
+            - An additional 'masks' is added, the non-binary masks with original size for the SPRO metric calculation
+        Args:
+            index (int): Index to get the item.
+
+        Returns:
+            dict[str, str | torch.Tensor]: Dict of image tensor during training. Otherwise, Dict containing image path,
+                target path, image tensor, label and transformed bounding box.
+        """
+        image_path = self.samples.iloc[index].image_path
+        mask_path = self.samples.iloc[index].mask_path
+        label_index = self.samples.iloc[index].label_index
+
+        image = read_image(image_path, as_tensor=True)
+        item = {"image_path": image_path, "label": label_index}
+
+        if self.task == TaskType.CLASSIFICATION:
+            item["image"] = self.transform(image) if self.transform else image
+        elif self.task in (TaskType.DETECTION, TaskType.SEGMENTATION):
+            # Only Anomalous (1) images have masks in anomaly datasets
+            # Therefore, create empty mask for Normal (0) images.
+            mask = (
+                Mask(torch.zeros(image.shape[-2:])).to(torch.uint8)
+                if label_index == LabelName.NORMAL
+                else read_mask(mask_path, as_tensor=True)
+            )
+            item["image"], item["mask"] = self.transform(image, mask) if self.transform else (image, mask)
+
+            item["mask_path"] = mask_path
+            # List of masks with the original size for saturation based metrics calculation
+            item["masks"] = mask
```

Comment:
Can we use a more descriptive name here? Not sure if this is clear from the first glance